src/wallet/bdb.h

```diff
@@ -161,6 +163,9 @@ class BerkeleyDatabase
     /** Database pointer. This is initialized lazily and reset during flushes, so it can be null. */
     std::unique_ptr<Db> m_db;
 
+    /** Make a BerkeleyBatch connected to this database */
+    std::unique_ptr<BerkeleyBatch> MakeBatch(const char* mode, bool flush_on_close);
```

Comment:
de110593c212a4198285e3f4966a7e2c14185867

I was expecting `std::unique_ptr<DatabaseBatch>` here - I suppose a future refactor will make `BerkeleyDatabase` extends `WalletDatabase` and that this method will be virtual?

Reply:
Yes

Reply:
Probably wouldn't hurt to do that in this PR.